x/fungible/keeper/deposits.go

```diff
@@ -142,17 +142,19 @@ func (k Keeper) processZetaDeposit(
 ) (*evmtypes.MsgEthereumTxResponse, bool, error) {
 	// Deposit/Mint ZETA to the protocol address which will then be used to make the calls below
 	// NOTE: DepositZeta and DepositAndCallZeta expect the fungible module account to have enough ZETA
-	if err := k.MintZetaToFungibleModule(ctx, amount); err != nil {
-		return nil, false, err
-	}
-
-	if isCrossChainCall {
-		res, err := k.DepositAndCallZeta(ctx, context, amount, to, message)
-		return res, true, err
-	}
-
-	res, err := k.DepositZeta(ctx, to, amount)
-	return res, false, err
+	return k.executeWithMintedZeta(
+		ctx,
+		amount,
+		func(tmpCtx sdk.Context) (*evmtypes.MsgEthereumTxResponse, bool, error) {
+			if isCrossChainCall {
+				res, err := k.DepositAndCallZeta(tmpCtx, context, amount, to, message)
+				return res, true, err
```

Comment:
nit but instead of true and false you can return isCrossChainCall for more readability?